### PR TITLE
Initial javascript implementation of cc validation in the UI (onblur)

### PR DESCRIPTION
**Pull request by borisson**
Initial js validation for credit cards.

### CODE REVIEW

**Comment by borisson** [comment]
I'll have to go trough the js again to add documentation and see if everything works as expected - but this is my initial work.

File: modules/payment/js/credit-card-validation.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+    return false;
+  };
+
+  /**
+   * Trigger all other validations.
+   *
+   * @param {object} element
+   */
+  var validation = function (element) {
```
Comment (bojanz) [inline modules/payment/js/credit-card-validation.js]: We need a better name for this method.

File: modules/payment/js/credit-card-validation.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    var value = element.val();
+    // Strip spaces from the value for all validations.
+    value = value.replace(/ /gi, '');
+
+    // If the value is not filled in, don't do any validation.
+    var empty_value = value.length === 0;
```
Comment (bojanz) [inline modules/payment/js/credit-card-validation.js:169]: If an element (such as CVV) exists, we should always validate that it's not empty. If it's empty, we want to show an error.

File: modules/payment/js/credit-card-validation.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    // Get the type of the card.
+    var type = detectType(value);
+
+    // If no type is found, don't bother doing anything else.
+    if (!type) {
+      return;
+    }
+
+    element.parent().parent().find('#cc-validation').html('CC is of type: ' + type.niceType);
```
Comment (bojanz) [inline modules/payment/js/credit-card-validation.js]: We need to figure out how to display the detected card type. Set a CSS class on some div? Perhaps we should look at what Amazon does.

**Comment by borisson** [comment]
As discussed, I now added an 'onkeyup' action that adds a class to the element with the card type.

File: modules/payment/js/credit-card-validation.js [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+      '5893',
+      '6304',
+      '6759',
+      '6761',
+      '6762',
+      '6763',
+      '0604',
+      '6390'
+    ],
+    lenghts: [12, 13, 14, 15, 16, 17, 18, 19]
```
Comment (steveoliver) [inline modules/payment/js/credit-card-validation.js:72]: typo: lengths

File: modules/payment/js/credit-card-validation.js [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+  var validateCreditCard = function (number, type) {
+    // Make sure that the type is really the expected type.
+    if (detectType(number) != type) {
+      return false;
+    }
+
+    // Test that the length of the card is actually one of the expected lengths
+    // defined in the type.
+    for (var x in type.lengths) {
+      var expected_lenght = type.lengths[x];
```
Comment (steveoliver) [inline modules/payment/js/credit-card-validation.js:149]: length typo in expected_length

File: modules/payment/js/credit-card-validation.js [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+      if (number.length === expected_lenght) {
```
Comment (steveoliver) [inline modules/payment/js/credit-card-validation.js:150]: length typo in expected_length

**Pull request closed without merging**